app/models/concerns/course/assessment/questions_concern.rb

```diff
@@ -71,6 +66,10 @@ def next_unanswered(submission)
   # @param [Course::Assessment::Submission] submission The submission which contains the answers.
   # @return [Array<Course::Assessment::Question>] The questions which were correctly answered.
   def correctly_answered_questions(submission)
-    where(id: submission.answers.where(correct: true).select(:question_id))
+    where(id: correctly_answered_question_ids(submision))
+  end
+
+  def correctly_answered_question_ids(submission)
```

Comment:
@lowjoel I have factored out the common parts of `not_correctly_answered` and `correctly_answered_questions`